src/auth/client.ts

```diff
@@ -60,17 +64,42 @@ export class ClientAuth {
       this.tokens.delete(hostname)
       return undefined
     }
-    return `${PeerIDAuthScheme} bearer="${token.bearer}"`
+    return { Authorization: `${PeerIDAuthScheme} bearer="${token.bearer}"`, peer: token.peer }
+  }
+
+  // authenticatedFetch is like `fetch`, but it also handles HTTP Peer ID
+  // authentication with the server.
+  //
+  // If we have not seen the server before, checkID will be called to check if
+  // we want to make the request to the server with the given peer id. This
+  // happens after we've authenticated the server.
```

Comment:
JSDoc-style comments will make this appear in the generated documentation:

```suggestion
  /**
   * authenticatedFetch is like `fetch`, but it also handles HTTP Peer ID
   * authentication with the server.
   *
   * If we have not seen the server before, checkID will be called to check if
   * we want to make the request to the server with the given peer id. This
   * happens after we've authenticated the server.
   */
```